apps/www/registry/default/ui/stepper.tsx

```diff
@@ -0,0 +1,421 @@
+    },
+  },
+  compoundVariants: [
+    {
+      isVertical: true,
+      isLastStep: true,
+      class: "w-full flex-[1_0_auto] flex-col items-start justify-start",
+    },
+  ],
+})
+
+export interface StepConfig extends StepLabelProps {
+  icon?: React.ReactElement
+}
+
+export interface StepProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof stepVariants>,
+    StepConfig {
+  isCompletedStep?: boolean
+}
+
+interface StepStatus {
+  index: number
+  isCompletedStep?: boolean
+  isCurrentStep?: boolean
+}
+
+interface StepAndStatusProps extends StepProps, StepStatus {}
+
+export const Step = React.forwardRef<HTMLDivElement, StepAndStatusProps>(
+  (props, ref) => {
+    const {
+      children,
+      description,
+      icon: CustomIcon,
+      index,
+      isCompletedStep,
+      isCurrentStep,
+      isLastStep,
+      label,
+      optional,
+      optionalLabel,
+    } = props
+
+    const {
+      isVertical,
+      isError,
+      isLoading,
+      successIcon: CustomSuccessIcon,
+      errorIcon: CustomErrorIcon,
+      isLabelVertical,
+      onClickStep,
+      isClickable,
+      variant,
+    } = useStepperContext()
+
+    const hasVisited = isCurrentStep || isCompletedStep
+
+    const handleClick = (index: number) => {
+      if (isClickable && onClickStep) {
+        onClickStep(index)
+      }
+    }
+
+    const Icon = React.useMemo(() => CustomIcon ?? null, [CustomIcon])
+
+    const Success = React.useMemo(
+      () => CustomSuccessIcon ?? <Check />,
+      [CustomSuccessIcon]
+    )
+
+    const Error = React.useMemo(
+      () => CustomErrorIcon ?? <X />,
+      [CustomErrorIcon]
+    )
+
+    const RenderIcon = React.useMemo(() => {
+      if (isCompletedStep) return Success
+      if (isCurrentStep) {
+        if (isError) return Error
+        if (isLoading) return <Loader2 className="animate-spin" />
+      }
+      if (Icon) return Icon
+      return (index || 0) + 1
+    }, [
+      isCompletedStep,
+      Success,
+      isCurrentStep,
+      Icon,
+      index,
+      isError,
+      Error,
+      isLoading,
+    ])
+
+    return (
+      <div
+        className={stepVariants({
+          isLastStep,
+          isVertical,
+          isClickable: isClickable && !!onClickStep,
+        })}
+        ref={ref}
+        onClick={() => handleClick(index)}
+        aria-disabled={!hasVisited}
+      >
+        <div
+          className={cn(
+            "flex items-center gap-2",
+            isLabelVertical ? "flex-col" : ""
+          )}
+        >
+          <Button
+            aria-current={isCurrentStep ? "step" : undefined}
+            data-invalid={isCurrentStep && isError}
+            data-highlighted={isCompletedStep}
+            data-clickable={isClickable}
+            disabled={!(hasVisited || isClickable)}
+            className={cn(
+              "h-12 w-12 rounded-full data-[highlighted=true]:bg-green-700 data-[highlighted=true]:text-white",
```

Comment:
Add ```aspect-square``` to the TailwindCSS to make the circle maintain its shape when the page is responsive.